lib/galaxy/schema/jobs.py

```diff
@@ -0,0 +1,268 @@
+from typing import (
+    Any,
+    Dict,
+    List,
+    Optional,
+    Union,
+)
+
+from pydantic import (
+    Extra,
+    Field,
+    Required,
+    UUID4,
+)
+
+from galaxy.schema.fields import (
+    DecodedDatabaseIdField,
+    EncodedDatabaseIdField,
+)
+from galaxy.schema.schema import (
+    DatasetSourceType,
+    EncodedDatasetSourceId,
+    EntityIdField,
+    JobSummary,
+    Model,
+    UuidField,
+)
+
+
+class JobInputSummary(Model):
+    has_empty_inputs: bool = Field(
+        default=Required,
+        title="Empty inputs",
+        description="Job has empty inputs.",
+    )
+    has_duplicate_inputs: bool = Field(
+        default=Required,
+        title="Duplicate inputs",
+        description="Job has duplicate inputs.",
+    )
+
+
+# TODO: Use Tuple again when `make update-client-api-schema` supports them
+class JobErrorSummary(Model):
+    # messages: List[Union[Tuple[str, str], List[str]]]
+    messages: List[List[str]] = Field(
+        default=Required,
+        title="Error messages",
+        description="The error messages for the specified job.",
+    )
+
+
+class JobAssociation(Model):
+    name: str = Field(
+        default=Required,
+        title="name",
+        description="The name of the associated dataset.",
+    )
+    dataset: EncodedDatasetSourceId = Field(
+        default=Required,
+        title="dataset",
+        description="The associated dataset.",
+    )
+
+
+class ReportJobErrorPayload(Model):
+    dataset_id: DecodedDatabaseIdField = Field(
+        default=Required,
+        title="Dataset ID",
+        description="The dataset ID related to the error.",
+    )
+    # TODO add description
+    email: Optional[str] = Field(
+        default=None,
+        title="Email",
+        description="",
+    )
+    message: Optional[str] = Field(
+        default=None,
+        title="Message",
+        description="The optional message sent with the error report.",
+    )
+
+
+class SearchJobsPayload(Model):
+    tool_id: str = Field(
+        default=Required,
+        title="Tool ID",
+        description="The tool ID related to the job.",
+    )
+    # TODO the inputs are actually a dict, but are passed as a JSON dump
+    # maybe change it?
+    inputs: str = Field(
+        default=Required,
+        title="Inputs",
+        description="The inputs of the job as a JSON dump.",
+    )
+    state: str = Field(
+        default=Required,
+        title="State",
+        description="The state of the job.",
+    )
+
+    class Config:
+        extra = Extra.allow  # This is used for items named file_ and __file_
+
+
+class DeleteJobPayload(Model):
+    message: Optional[str] = Field(
+        default=None,
+        title="Job message",
+        description="Stop message",
+    )
+
+
+class EncodedDatasetJobInfo(EncodedDatasetSourceId):
+    uuid: UUID4 = UuidField
```

Comment:
Same comment applies, this can be more than just HistoryDatasetAssociation and LibraryDatasetDatasetAssociation, so `EncodedDatasetSourceId` is not correct, and `uuid` is optional (and would love if you can mark this as deprecated, it wasn't well-thought through when it was added to the API).

Reply:
So like this?
```

uuid: Optional[UUID4] = Field(
        default=None,
        deprecated=True,
        title="UUID",
        description="Universal unique identifier for this dataset. In this context the uuid is optional and marked as deprecated.",
    )
```

Reply:
```
uuid: Optional[UUID4] = Field(
        default=None,
        deprecated=True,
        title="UUID",
        description="Universal unique identifier for this dataset.",
    )
```
Optional and deprecated should be clear from the schema, don't need to repeat that.